testing-commandments.md

```diff
@@ -0,0 +1,30 @@
+# 8 Commandments of Automated Testing
+
+*DISCLAIMER* - Failure to meet any of these points doesn't mean new ideas or pull requests should be outright rejected. They should be used as points of discussion to ultimately help us ship code faster.
+
+`code` is used in reference to the code being tested. The target, maybe?
+`test` is used in reference to the code that is exercising the codebase to validate correctness.
+
+## Thou shalt be consistent within projects
+If a project has a de-facto standard, it should be maintained. The process for changing a de-facto standard should be people focused, and should not be initially communicated via a pull request.
+
+## Thy tests should be obvious
+If the test behaviour isn't obvious by looking at it, it's no good. If I have to scroll for days to read a test, it's no good.
+
+## Thy tests should only check for something once
+If I make a single code change and multiple tests fail, it's no good.
```

Comment:
I think get what you're trying to say here, but I don't think this sentence does a good job at explaining that. If we were strictly unit testing everything then this would make sense, but with any level of integration testing a single code change may break many tests. especially if it's in a shared module/library reused several times in a larger project.

I interpreted the message behind this as "Tests cost money to write, understand, maintain and run. So don't double up on testing the same thing several times" . Perhaps an alternative description could be _"If a test suite checks the same behaviour several times, it's no good"_?